lib/puppet/pops/evaluator/runtime3_support.rb

```diff
@@ -298,14 +298,11 @@ def external_call_function(name, args, scope, &block)
   def call_function(name, args, o, scope, &block)
     file, line = extract_file_line(o)
     loader = Adapters::LoaderAdapter.loader_for_model_object(o, file)
-    # 'ruby -wc' thinks that _func is unused, because the only reference to it
-    # is inside of the Kernel.eval string below. By prefixing it with the
-    # underscore, we let Ruby know to not worry about whether it's unused or not.
-    _func = loader.load(:function, name) if loader
-    if _func
+    func = loader.load(:function, name) if loader
+    if func
       Puppet::Util::Profiler.profile(name, [:functions, name]) do
-        # Add stack frame when calling. See Puppet::Pops::PuppetStack
-        return Kernel.eval('_func.call(scope, *args, &block)'.freeze, Kernel.binding, file || '', line)
```

Comment:
This was moved to not call PuppetStack.stack in PUP-6622 as a performance improvement on Windows. While I think this work in general should provide a meaningful performance speed up in JRuby, we should validate and that there wasn't a regression in Windows.